internal/controllers/control/cassandratask_controller_test.go

```diff
@@ -622,10 +622,25 @@ var _ = Describe("CassandraTask controller tests", func() {
 				Expect(callDetails.URLCounts["/api/v0/ops/executor/job"]).To(Equal(0))
 				Expect(callDetails.URLCounts["/api/v0/metadata/versions/features"]).To(BeNumerically(">", 1))
 
+				// verifyPodsHaveAnnotations(testNamespaceName, string(task.UID))
+				Expect(completedTask.Status.Succeeded).To(BeNumerically(">=", 1))
+			})
+			It("Runs a ts reload task against a pod", func() {
```

Comment:
issue: Well, this is not actually broken in the implementation, but this test doesn't actually verify your Sync check is working. As seen here, the Feature is not in the list of available Features: 

https://github.com/k8ssandra/cass-operator/blob/3e146206ec9493e1ec87d018ea3becae27d799c2/pkg/httphelper/server_test_utils.go#L15

But this succeeds, because the test is ran against a fake server without details endpoint. And because your SyncFeature is not set in the jobs.go.

Reply:
Interesting, thanks. I will investigate this some more, it seemed like this was too easy,.